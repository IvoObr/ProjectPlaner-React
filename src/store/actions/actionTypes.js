export const actionTypes = {
    START_FETCH_PROJECTS: 'START_FETCH_PROJECTS',
    START_FETCH_PROJECTS_SUCCESS: 'START_FETCH_PROJECTS_SUCCESS',
    START_FETCH_PROJECTS_FAILED: 'START_FETCH_PROJECTS_FAILED',
    CREATE_PROJECT: 'CREATE_PROJECT',
    CREATE_PROJECT_SUCCESS: 'CREATE_PROJECTS_SUCCESS',
    CREATE_PROJECT_FAILED: 'CREATE_PROJECT_FAILED',

}