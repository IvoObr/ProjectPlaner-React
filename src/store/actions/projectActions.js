import {url, requestObject, responseHandle} from '../fetchConfig'

export const createProject = (project) => {
    return(dispatch, getState) => {
        let bodyObj = {
            ...project,
        };

        requestObject.method =  'POST';
        requestObject.body = JSON.stringify(bodyObj)
        fetch(`${url}/projects`, requestObject)
            .then(response => {
                responseHandle(response, dispatch, 'CREATE_PROJECT');
            }).catch(error => {
            console.log(new Error(`Request rejected: ${error}`));
        });
    }
};

export const getProjects = () => {
    return(dispatch, getState) => {
        requestObject.method =  'GET';
        fetch(`${url}/projects`, requestObject)
            .then(response => {
                responseHandle(response, dispatch, 'GET_ALL_PROJECTS');
            }).catch(error => {
            console.log(new Error(`Request rejected: ${error}`));
        });
    }
};