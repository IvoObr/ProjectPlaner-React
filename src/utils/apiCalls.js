import axios from 'axios';

const url = 'http://localhost:3002';

export const fetchProjects = () => axios.get(`${url}/projects`).then(response => response.data);
export const createProject = (body) => axios.post(`${url}/projects`, body).then(response => response.data);


