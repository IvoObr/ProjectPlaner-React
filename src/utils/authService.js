import axios from 'axios';
import url from './serviceConfing'

const saveTokenInSession = (response) => {
    const xAuth = response.request.getResponseHeader('x-auth');
    response.data.user.token = xAuth;
    localStorage.setItem('user', JSON.stringify(response.data.user));
};

export const login = (user) => {
    return axios.post(`${url}/users/login`, user).then(response => {
        saveTokenInSession(response);
        return response.data
    });
};

export const logout = () => {
    const token = localStorage.getItem('user') ? JSON.parse(localStorage.getItem('user')).token : null;
    const headers = {
        'x-auth': token || null
    };

    return axios.delete(`${url}/users/token`, {headers}).then(response => {
        localStorage.removeItem('user');
        return response.data
    });
};

export const signup = (user) => {
    return axios.post(`${url}/users`, user).then(response => {
        saveTokenInSession(response);
        return response.data
    });
};