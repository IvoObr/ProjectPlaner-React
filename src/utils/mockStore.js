export const mockStore = {
    auth: {
        loggingIn: false,
        loggedIn: true,
        user: {
            _id: '5c643ebfe0a0aa4ca8efd0e2',
            email: 'kiro_bora@yahoo.com',
            firstName: 'kiro',
            lastName: 'bora',
            initials: 'kb',
            token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1YzY0M2ViZmUwYTBhYTRjYThlZmQwZTIiLCJhY2Nlc3MiOiJhdXRoIiwiaWF0IjoxNTUyOTk3ODkyLCJleHAiOjE1NTMwODQyOTJ9.Zp0F_UK8y9Jr48rp6aUPvgIjFNsHOLVmY9t_XUdSymY'
        },
        loginError: ''
    },
    project: {
        projects:  [
            {
                createdAt: '2019-03-05T19:59:07.156Z',
                _id: '5c7ed57413f8e60017102a23',
                title: 'proj title 2',
                content: 'test project text',
                authorFirstName: 'kiro',
                authorLastName: 'bora',
                authorId: '5c643ebfe0a0aa4ca8efd0e2',
                __v: 0
            },
            {
                createdAt: '2019-03-05T19:59:07.156Z',
                _id: '5c7ed57a13f8e60017102a24',
                title: 'proj title',
                content: 'test project text 2',
                authorFirstName: 'kiro',
                authorLastName: 'bora',
                authorId: '5c643ebfe0a0aa4ca8efd0e2',
                __v: 0
            }
        ],
        isProjectCreated: false,
        isProjectSaved: false
    }
};
