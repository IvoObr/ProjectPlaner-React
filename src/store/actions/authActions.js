import {actionTypes} from "./actionTypes";

export const actions = {
  login: (user) => ({
    type: actionTypes.LOGIN_USER,
    payload: {user}
  }),
  loginSuccess: (user) => ({
    type: actionTypes.LOGIN_USER_SUCCESS,
    payload: {user}
  }),
  loginFailed: (user) => ({
    type: actionTypes.LOGIN_USER_FAILED,
    payload: {user}
  }),
  logout: (user) => ({
    type: actionTypes.LOGOUT_USER,
    payload: {user}
  }),
  logoutSuccess: (user) => ({
    type: actionTypes.LOGOUT_USER_SUCCESS,
    payload: {user}
  }),
  logoutFailed: (user) => ({
    type: actionTypes.LOGOUT_USER_FAILED,
    payload: {user}
  }),
  signup: (user) => ({
    type: actionTypes.SIGNUP_USER,
    payload: {user}
  }),
  signupSuccess: (user) => ({
    type: actionTypes.SIGNUP_USER_SUCCESS,
    payload: {user}
  }),
  signupFailed: (user) => ({
    type: actionTypes.SIGNUP_USER_FAILED,
    payload: {user}
  }),
};