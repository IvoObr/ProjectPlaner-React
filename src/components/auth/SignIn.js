import React, {Component} from 'react';
import {connect} from "react-redux";
import {actions} from "../../store/actions/authActions";
import {Redirect} from "react-router-dom";
import Notifications from "./Notifications";

class SignIn extends Component {
    state = {
        email: '',
        password: ''
    };

    handleChange = (event) => {
        this.setState({
            [event.target.id]: event.target.value
        })
    };

    handleSubmit = (event) => {
        event.preventDefault();
        this.props.login(this.state);
    };

    render() {
        if (this.props.isLoggedIn) {
           return <Redirect to='/'/>
        }

        return (
            <div className='container'>
                <form onSubmit={this.handleSubmit} className='white'>
                    <h5 className="grey-text text-darken-3">Login</h5>

                    <div className="input-field">
                        <label htmlFor="email">Email</label>
                        <input type="email" id='email' onChange={this.handleChange}/>
                    </div>

                    <div className="input-field">
                        <label htmlFor="password">Password</label>
                        <input type="password" id='password' onChange={this.handleChange}/>
                    </div>

                    <div className="input-field">
                        <button className="btn green lighten-1 z-depth-0">Login</button>
                    </div>
                        {this.props.loggingIn ? <div className="progress"> <div className="indeterminate"></div> </div> : <div className="height-19"></div>}
                        <Notifications doShow={this.props.loginError} message={'Invalid username or password'}/>
                    </form>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
      return {
          isLoggedIn: state.auth.loggedIn,
          loginError: state.auth.loginError,
          loggingIn: state.auth.loggingIn,
      }
};

const mapDispatchToProps = {
    login: actions.login
};

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);