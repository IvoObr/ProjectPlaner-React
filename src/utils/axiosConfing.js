import axios from 'axios';
import {store} from '../index';
import {actions} from '../store/actions/authActions';

axios.interceptors.response.use((response) => {
    return response;
}, (error) => {
    if (401 === error.response.status) {
        localStorage.removeItem('user');
        store.dispatch(actions.logout());
        return Promise.reject('Unauthorized');
    } else {
        return Promise.reject(error);
    }
});

export default 'https://project-planer-node-js.herokuapp.com';