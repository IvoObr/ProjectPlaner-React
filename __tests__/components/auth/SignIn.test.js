import React from 'react';
import ReactDOM from 'react-dom';
import SignIn from '../../../src/components/auth/SignIn';
import {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import {actions} from '../../../src/store/actions/authActions';
import {mockStore} from '../../../src/utils/mockStore'

configure({adapter: new Adapter()});

describe('<SignIn/>', () => {
    let wrapper = null;
    let props = null;
    let store = {};

    beforeEach(() => {
        store = configureStore(mockStore);
        props = {
            login: actions.login,
            isLoggedIn: false
        };

        wrapper = shallow(<SignIn.WrappedComponent {...props} store={store}/>);
    });

    it('should render SignIn', () => {
        expect(wrapper.find('div').length).toBe(4);
    });

    it('matches snapshot', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('handleChange should work', () => {
        let event = {
            target: {
                id: 'email',
                value: 'test'
            }
        };

        wrapper.instance().handleChange(event);
        let stateEmail = wrapper.state().email;
        expect(stateEmail).toBe('test');
    });

    it('handleSubmit should work', () => {
        let event = {
            preventDefault() {
                return true;
            }
        };

        const throwError = wrapper.instance().handleSubmit(event);
        expect(throwError).toBeUndefined();
    });

    it('should redirect if logged in redirect to Home', () => {
        props.isLoggedIn = true;
        wrapper = shallow(<SignIn.WrappedComponent {...props}/>);
        expect(wrapper.find('Redirect').length).toBe(1);
    });
});
