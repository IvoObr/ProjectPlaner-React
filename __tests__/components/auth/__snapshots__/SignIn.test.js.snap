// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn/> matches snapshot 1`] = `
<div
  className="container"
>
  <form
    className="white"
    onSubmit={[Function]}
  >
    <h5
      className="grey-text text-darken-3"
    >
      Login
    </h5>
    <div
      className="input-field"
    >
      <label
        htmlFor="email"
      >
        Email
      </label>
      <input
        id="email"
        onChange={[Function]}
        type="email"
      />
    </div>
    <div
      className="input-field"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        id="password"
        onChange={[Function]}
        type="password"
      />
    </div>
    <div
      className="input-field"
    >
      <button
        className="btn green lighten-1 z-depth-0"
      >
        Login
      </button>
    </div>
    <Loading />
    <Notifications
      message="Invalid username or password"
    />
  </form>
</div>
`;
